ownCloudAppShared/Intent/GetFileIntentHandler.swift

```diff
@@ -0,0 +1,126 @@
+//
+//  GetFileIntentHandler.swift
+//  ownCloudAppShared
+//
+//  Created by Matthias Hühne on 30.07.19.
+//  Copyright © 2019 ownCloud GmbH. All rights reserved.
+//
+
+/*
+ * Copyright (C) 2019, ownCloud GmbH.
+ *
+ * This code is covered by the GNU Public License Version 3.
+ *
+ * For distribution utilizing Apple mechanisms please see https://owncloud.org/contribute/iOS-license-exception/
+ * You should have received a copy of this license along with this program. If not, see <http://www.gnu.org/licenses/gpl-3.0.en.html>.
+ *
+ */
+
+import UIKit
+import Intents
+import ownCloudSDK
+
+@available(iOS 13.0, watchOS 6.0, *)
+typealias GetFileCompletionHandler = (GetFileIntentResponse) -> Void
+
+@available(iOS 13.0, watchOS 6.0, *)
+public class GetFileIntentHandler: NSObject, GetFileIntentHandling {
+
+	var core : OCCore?
+	var completion : GetFileCompletionHandler?
+	var itemTracking : OCCoreItemTracking?
+
+	public func handle(intent: GetFileIntent, completion: @escaping (GetFileIntentResponse) -> Void) {
+
+		if AppLockHelper().isPassCodeEnabled {
+			completion(GetFileIntentResponse(code: .authenticationRequired, userActivity: nil))
+		} else {
+			if let path = intent.path, let uuid = intent.account?.uuid {
+				let accountBookmark = OCBookmarkManager.shared.bookmark(for: uuid)
+
+				if let bookmark = accountBookmark {
+					OCCoreManager.shared.requestCore(for: bookmark, setup: nil, completionHandler: { (core, error) in
+						if error == nil {
+							self.core = core
+							self.itemTracking = self.core?.trackItem(atPath: path, trackingHandler: { (error, item, isInitial) in
+								if let item = item {
+									if core?.localCopy(of: item) == nil {
+										core?.downloadItem(item, options: [ .returnImmediatelyIfOfflineOrUnavailable : true ], resultHandler: { (error, core, item, file) in
+											if error == nil {
```

Comment:
@hosy a lot of nesting levels makes code not easy to follow.. Can it be simplified?

Reply:
I removed some nesting levels, but most are needed for specific failure code